modules/reactor/src/meshgenerators/RevolveGenerator.C

```diff
@@ -0,0 +1,2340 @@
+            else if (nodes_cates.first.size() == 3)
+            {
+              createTETfromTRI(nodes_cates,
+                               TET10,
+                               elem,
+                               mesh,
+                               new_elem,
+                               current_layer,
+                               orig_nodes,
+                               total_num_azimuthal_intervals,
+                               side_pairs,
+                               axis_node_case,
+                               isFlipped);
+            }
+            else
+              mooseError("impossible situation");
+            break;
+          }
+          case TRI7:
+          {
+            // Possible scenarios:
+            // 1. None of the nodes are on the axis
+            //    Then a prism21 element is created
+            // 2. One of the nodes is on the axis
+            //    Then a pyramid18 element is created
+            // 3. Three of the nodes are on the axis
+            //    Then a tet14 element is created
+            // NOTE: We do not support two nodes on the axis for tri7 elements
+            const auto nodes_cates = onAxisNodesIdentifier(*elem, nodes_on_axis);
+            if (nodes_cates.first.empty())
+            {
+              createPRISMfromTRI(PRISM21,
+                                 elem,
+                                 mesh,
+                                 new_elem,
+                                 current_layer,
+                                 orig_nodes,
+                                 total_num_azimuthal_intervals,
+                                 side_pairs,
+                                 isFlipped);
+            }
+            else if (nodes_cates.first.size() == 1)
+            {
+              createPYRAMIDfromTRI(nodes_cates,
+                                   PYRAMID18,
+                                   elem,
+                                   mesh,
+                                   new_elem,
+                                   current_layer,
+                                   orig_nodes,
+                                   total_num_azimuthal_intervals,
+                                   side_pairs,
+                                   axis_node_case,
+                                   isFlipped);
+            }
+            else if (nodes_cates.first.size() == 3)
+            {
+              createTETfromTRI(nodes_cates,
+                               TET14,
+                               elem,
+                               mesh,
+                               new_elem,
+                               current_layer,
+                               orig_nodes,
+                               total_num_azimuthal_intervals,
+                               side_pairs,
+                               axis_node_case,
+                               isFlipped);
+            }
+            else
+              mooseError("impossible situation");
```

Comment:
mention the element is either degenerate or unsupported, based on the node_cates